src/pages/index.js

```diff
@@ -21,36 +37,134 @@ const imageModal = new ModalWithImage({
   modalSelector: "#view-image-modal",
 });
 
+const deleteModal = new ModalDelete({
+  modalSelector: "#delete-image-modal",
+  cardRemovalHandler: handleCardRemoval,
+  card: undefined,
+});
+
 // profile section
 
 const profileEl = new UserInfo({
   nameSelector: "#profile-name",
   descriptionSelector: "#profile-description",
+  avatarSelector: "#avatar",
 });
 
 // submission handlers
 
 function handleProfileEditSubmit(e, inputValues) {
   e.preventDefault();
+  consts.profileEditDiv.querySelector(".modal__button").innerText = "Saving...";
   const name = inputValues["Name"];
   const description = inputValues["Description"];
-  profileEl.setUserInfo(name, description);
-  profileEditValidator.disableSubmitButton();
+  return api
+    .updateProfileInfo(name, description)
+    .then(() => {
+      profileEl.setUserInfo(name, description);
+      formValidation[1].disableSubmitButton();
+    })
+    .finally(() => {
+      profileEditModal.close();
```

Comment:
You need to close popups only in blocks `then` after a **successful response** from the server so that when there is an error it would be opened (the inputs will not be cleared and the user could send the data again to the server) and if you change the button text you could see the change as well